### PR TITLE
Add vtunnel forwarder

### PR DESCRIPTION
**Pull request by Nino-K**
This PR adds a forwarding process to Vtunnel peer.

### CODE REVIEW

File: pkg/forwarder/vtunnel.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package forwarder
+
+import (
+	"encoding/json"
+	"net"
+
+	"github.com/rancher-sandbox/rancher-desktop-agent/pkg/types"
+)
+
+const vtunnelPeerAddr = "127.0.0.1:3040"
```
Comment (mook-as) [inline pkg/forwarder/vtunnel.go]: We'll need to be able to thread this in from a command line argument… At least the port.

**Review by mook-as — CHANGES_REQUESTED** [review]
Do we have an OpenAPI spec or something for the privileged service? That will be necessary soon…

File: cmd/rancher-desktop-guestagent/main.go [hunk 2]
```diff
@@ -68,17 +71,23 @@ func main() {
...
+
 	if *enableDocker {
+		if *vtunnelAddr == "" {
+			log.Fatal("vtunnel address must be provided when docker is enable.")
+		}
+
 		group.Go(func() error {
-			portTracker := tracker.NewPortTracker()
-			eventMonitor, err := docker.NewEventMonitor()
+			forwarder := forwarder.NewVtunnelForwarder((*vtunnelAddr))
```
Comment (mook-as) [inline cmd/rancher-desktop-guestagent/main.go]:
```suggestion
			forwarder := forwarder.NewVtunnelForwarder(*vtunnelAddr)
```
Unless there was some particular reason for the extra parentheses…

**Review by mook-as — APPROVED** [review]

File: cmd/rancher-desktop-guestagent/main.go (hunk 2 above)
Comment (Nino-K) [reply cmd/rancher-desktop-guestagent/main.go]: I'm surprised gofmt and glalngci-lint didn't catch that 🤔

**Comment by Nino-K** [comment]
https://github.com/rancher-sandbox/rancher-desktop/issues/2677

**Pull request merged**